.github/workflows/test-backend.yml

```diff
@@ -0,0 +1,47 @@
+name: Test Backend
+on: 
+  pull_request:
+
+jobs:
+  testing:
+    name: Testing Backend
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./backend
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          pip install pytest-md pytest-emoji
+
+        # some tests require an .env file to run.
+        # file should contain all the required properties.  
+      - name: Create environment file
+        run: |
+          echo "Create environment file"
+          cat <<EOF > .env
+          MODEL=my-model
+          MISTRAL_KEY=123456
+          NEO4J_URI=bolt://localhost:1234
+          NEO4J_USERNAME=neo4j
+          NEO4J_PASSWORD=my-password
+          EOF
+          cat .env
+
+      - name: Run tests
+        uses: pavelzw/pytest-action@v2
+        with:
+          emoji: true
+          verbose: true
```

Comment:
ah I'm guessing this is where those requirements are needed? Is this action better than just running `pytest` without a marketplace action?

Reply:
The major benefit is the reporting. You get a nice summary in markdown. I'll give you a demo

Reply:
Thanks for the demo, the reports look really cool